lib/pipelineTemplateVersionFactory.js

```diff
@@ -0,0 +1,206 @@
+     * @param  {String}     config.description   Description of the template
+     * @param  {String}     config.maintainer    Maintainer's email
+     * @param  {Object}     config.config        Config of the screwdriver-template.yaml
+     * @param  {String}     config.pipelineId    pipelineId of the template
+     * @param  {Array}      [config.labels]      Labels attached to the template
+     * @return {Promise}
+     */
+    // eslint-disable-next-line no-shadow
+    create(config, pipelineTemplateFactory) {
+        // 0. create variable createTime: result.createTime = new Date().toISOString();
+        // 1. get template meta using template meta factory which matches name and namespace
+        // 2. if template meta doesn't exist, create template meta with all necessary fields, createTime and updateTime is identical
+        // 3. get version from version factory for the specified major,minor version from the config
+        //      if version from config is exactVersion, make a DB call with below parameters to get the version entry and return it:
+        //          templateId from templateMeta (from steps 1, 2)
+        //          version from config
+        //      else return [pipelineTemplate, NULL]
+        // 4. create a new version in version factory by passing necessary fields
+        //      fields coming from config= {description, config}
+        //      templateId from templateMeta (from steps 1, 2)
+        //      createTime from step 0
+        //      need to determine exact version (may only have part of the version initially):
+        //          if step 3 returns NULL:
+        //              if we have exact version from config, use exact version
+        //              if we have partial version:
+        //                  if global version is NULL, complete the partial version and use it
+        //                  else bump up global version and use it
+        //          else bump up exact version and use it
+        // 5. update latestVersion in templateMeta (if it is overall latest) -> update time: result.updateTime = new Date().toISOString();
+
+        const createTime = new Date().toISOString();
+
+        pipelineTemplateFactory
+            .get({
+                params: {
+                    name: config.name,
+                    namespace: config.namespace
+                }
+            })
+            // eslint-disable-next-line no-shadow
+            .then(pipelineTemplate => {
+                if (!pipelineTemplate) {
+                    return pipelineTemplateFactory.create({
+                        pipelineId: config.pipelineId,
+                        namespace: config.namespace,
+                        name: config.name,
+                        maintainer: config.maintainer,
+                        createTime,
+                        updateTime: createTime,
+                        templateType: 'PIPELINE'
```

Comment:
`templateType` should be set in `TemplateMetaFactory` by calling `getTemplateType()`
```suggestion
                        updateTime: createTime
```